_test/test_utilities_herbert/test_str_to_iarray.m

```diff
@@ -0,0 +1,187 @@
+            iarr = str_to_iarray ('37');
+            assertEqual (iarr, 37)
+        end
+        
+        %--------------------------------------------------------------------------
+        % Matlab style tokens
+        %--------------------------------------------------------------------------
+        function test_range (~)
+            iarr = str_to_iarray ('-3:6');
+            assertEqual (iarr, -3:6)
+        end
+        
+        function test_range_rhs_lt_lhs (~)
+            iarr = str_to_iarray ('6:-3');
+            assertEqual (iarr, 6:-3)
+        end
+        
+        function test_range_with_positive_stride (~)
+            iarr = str_to_iarray ('3:2:8');
+            assertEqual (iarr, 3:2:8)
+        end
+        
+        function test_range_with_negative_stride (~)
+            iarr = str_to_iarray ('3:-2:8');
+            assertEqual (iarr, 3:-2:8)
+        end
+        
+        function test_range_with_positive_stride_rhs_lt_lhs (~)
+            iarr = str_to_iarray ('3:2:-4');
+            assertEqual (iarr, 3:2:-4)
+        end
+        
+        function test_range_with_negative_stride_rhs_gt_lhs (~)
+            iarr = str_to_iarray ('3:-2:8');
+            assertEqual (iarr, 3:-2:8)
+        end
+
+        function test_range_with_zero_stride_rhs_lt_lhs (~)
+            func = @()str_to_iarray ('3:0:-4');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+        
+        function test_range_with_zero_stride_rhs_gt_lhs (~)
+            func = @()str_to_iarray ('3:0:8');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+        
+        
+        %--------------------------------------------------------------------------
+        % non-Matlab format tokens
+        %--------------------------------------------------------------------------
+        function test_non_Matlab_range_pos2pos_ascend (~)
+            iarr = str_to_iarray ('3-8');
+            assertEqual (iarr, 3:8)
+        end
+        
+        function test_non_Matlab_range_pos2pos_descend (~)
+            iarr = str_to_iarray ('8-3');
+            assertEqual (iarr, 8:-1:3)
+        end
+        
+        function test_non_Matlab_range_neg2pos_ascend (~)
+            iarr = str_to_iarray ('-3-8');
+            assertEqual (iarr, -3:8)
+        end
+        
+        function test_non_Matlab_range_pos2neg_descend (~)
+            iarr = str_to_iarray ('3--8');
+            assertEqual (iarr, 3:-1:-8)
+        end
+        
+        function test_non_Matlab_range_neg2neg_ascend (~)
+            iarr = str_to_iarray ('-8--3');
+            assertEqual (iarr, -8:-3)
+        end
+        
+        function test_non_Matlab_range_neg2neg_descend (~)
+            iarr = str_to_iarray ('-3--8');
+            assertEqual (iarr, -3:-1:-8)
+        end
+       
+        function test_non_Matlab_range_neg2neg_single (~)
+            % Tough test of parsing?
```

Comment:
when the non matlab ranges get tough, the horace parsing gets going

Reply:
Actually I've driven myself half-mad with parsing over the past two weeks. I'm only dealing with the non-Matlab forms for reasons of legacy code.